### PR TITLE
fix(CookieStore): check if "localStorage" is defined before using it

### LINKED ISSUE #7: `CookieStore.hydrate` fails when running in Node.js env because `localStorage` is undefined
I am  using `setupServer` of `msw/node` to mock endpoint for tests. 

I got the following error: 

```
    TypeError: Cannot read property '_origin' of null

      at Window.get localStorage [as localStorage] (../../../../node_modules/jsdom/lib/jsdom/browser/Window.js:420:50)
      at CookieStore.hydrate (../../../../node_modules/msw/node/lib/index.js:7448:34)
      at setRequestCookies (../../../../node_modules/msw/node/lib/index.js:7549:17)
      at ../../../../node_modules/msw/node/lib/index.js:7609:21
      at ../../../../node_modules/msw/node/lib/index.js:53:71
      at __awaiter (../../../../node_modules/msw/node/lib/index.js:49:12)
      at resolver (../../../../node_modules/msw/node/lib/index.js:7583:24)
      at ClientRequestOverride.<anonymous> (../../../../node_modules/@mswjs/interceptors/src/interceptors/ClientRequest/createClientRequestOverride.ts:189:9)
```

The error occured because the `localStorage` is undefined. 

I know I can mock it, but it will be good that we don't have to.

BTW, thank you for all of `msw/*` packages, very usefull

### PR DESCRIPTION
**Pull request by neilime**
Fixes #7 

- Check if `localStorage` is defined before using it.
- Prettier format

### CODE REVIEW

File: src/CookieStore.ts [hunk 1]
```diff
@@ -186,6 +193,14 @@ Invalid value has been removed from localStorage to prevent subsequent failed pa
       }
     })
   }
+
+  /**
+   * Retrieve the global local storage if defined
+   * @returns Storage | undefined
+   */
+  private getLocalStorage(): Storage | undefined {
```
Comment (kettanaito) [inline src/CookieStore.ts:201]: I don't think we get any benefits of keeping this as a private method. We'd rather store it as a module-scoped variable, as `localStorage` is either defined or not (determined by the environment). Getter makes sense if the storage is eventually defined, but this is not the case.

File: src/CookieStore.ts [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
 interface ResponseLike {
-    headers: HeadersLike;
+  headers: HeadersLike
 }
 type Store = Map<string, StoreEntry>
 type StoreEntry = Map<string, Cookie>
-type CookieString = Omit<Cookie, 'expires'> & { expires?: string };
+type CookieString = Omit<Cookie, 'expires'> & { expires?: string }
 
 export const PERSISTENCY_KEY = 'MSW_COOKIE_STORE'
```
Comment (kettanaito) [inline src/CookieStore.ts:18]: I'd define the global variable here, determining if we can use local storage:

```js
const SUPPORTS_LOCAL_STORAGE = typeof localStorage !== 'undefined'
```

**Review by kettanaito — CHANGES_REQUESTED** [review]
Thank you for working on this, @neilime! Please, could you look into the comments I left?
It'd be great to separate the prettier changes from the actual fix, so we have a cleaner Git history. Thanks!

File: src/CookieStore.ts (hunk 2 above)
Comment (neilime) [reply src/CookieStore.ts:18]: If the localStorage is removed during runtime, it will fail. Using a getter avoid this case

Comment (kettanaito) [reply src/CookieStore.ts:18]: What is the case when the `localStorage` is removed during runtime? I'd say that's an edge-case scenario that indicates an error in the runtime, but I may be wrong.

File: test/persistency.test.ts [hunk 3]
```diff
@@ -45,6 +46,36 @@ test('persists cookies in localStorage', () => {
...
+    const req = new Request('https://mswjs.io')
+    const res = new Response(null, {
+      headers: new Headers({ 'set-cookie': 'cookieName=abc-123' }),
+    })
+
+    store.add(req, res)
+    store.persist()
+
+    localStorage = originalLocalStorage
+    expect(localStorage.getItem(PERSISTENCY_KEY)).toEqual(null)
```
Comment (kettanaito) [inline test/persistency.test.ts:79]: Despite the cookies not being persisted in the local storage, I think we should still assert that they persist in the internal `this.store` map when you try to retrieve them.

Comment (neilime) [reply test/persistency.test.ts:88]: Done @kettanaito

File: src/CookieStore.ts (hunk 2 above)
Comment (neilime) [reply src/CookieStore.ts:18]: I don't have any idea of what can be the case.

But this is technically possible.

By the way, if we store the check in a const as you recommend, I'll need extra time to find a way to test undefined localStorage. Because it cannot work as it is currently done

**Comment by neilime** [comment]
:up:  :smile:

**Review by kettanaito — APPROVED** [review]
Thank you for shipping this improvement, @neilime!
